semantic-analysis/src/Analysis/Carrier/Store/Monovariant.hs

```diff
@@ -83,3 +87,15 @@ instance Has (State (MStore value)) sig m
         pure (MAddr n <$ Map.lookup (MAddr n) store <$ ctx)
 
     R other -> EnvC (alg (runEnv . hdl) other ctx)
+
+
+-- Running
+
+runFiles
+  :: (forall sig m . Has (State (MStore  value)) sig m => File term -> m (File result))
+  -> [File term]
+  -> (MStore value, [File result])
+runFiles runFile
+  = run
+  . runStoreState
+  . traverse runFile
```

Comment:
This convenience wraps up a pattern common among the (0-CFA) analyses into a convenient bundle. It's also a great example of exposing effects while hiding carriers, the essence of abstraction.